matplotcheck/tests/test_base_data.py

```diff
@@ -72,6 +83,143 @@ def test_assert_xydata_xlabel_fails(pt_bar_plt, pd_df):
     plt.close()
 
 
+def test_assert_xydata_xlabel_text():
+    "Tests the xlabels flag on xydata works to test labels with text data"
+    data = {
```

Comment:
same here -- use fixtures rather than recreating objects over and over. the tests will be greatly simplified then!